lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -872,4 +872,39 @@ previewers.display_content = defaulter(function(_)
   }
 end, {})
 
+previewers.buffers = defaulter(function(opts)
+  opts = opts or {}
+  local cwd = opts.cwd or vim.loop.cwd()
+  return Previewer:new {
+    title = function()
+      return "Buffers"
+    end,
+    dyn_title = function(_, entry)
+      return Path:new(from_entry.path(entry, true)):normalize(cwd)
+    end,
+    preview_fn = function(self, entry, status)
+      if vim.api.nvim_buf_is_valid(entry.bufnr) then
+        vim.api.nvim_win_set_buf(status.preview_win, entry.bufnr)
+        vim.api.nvim_win_set_option(status.preview_win, "winhl", "Normal:TelescopePreviewNormal")
+        vim.api.nvim_win_set_option(status.preview_win, "signcolumn", "no")
+        vim.api.nvim_win_set_option(status.preview_win, "foldlevel", 100)
+        vim.api.nvim_win_set_option(status.preview_win, "wrap", false)
+        self.state.bufnr = entry.bufnr
+      end
+    end,
+    scroll_fn = function(self, direction)
+      if not self.state then
+        return
+      end
+
+      local input = direction > 0 and [[]] or [[]]
+      local count = math.abs(direction)
+
+      vim.api.nvim_buf_call(self.state.bufnr, function()
+        vim.cmd([[normal! ]] .. count .. input)
+      end)
+    end,
+  }
```

Comment:
We should reuse this. So local function and then just setting it in `new_buffer_previewer` and here.

Reply:
This is a lot more funny than I've thought. It has the side effect that it scrolls the actual buffer in the background when pressing <c-u> for `builtin.buffers` :laughing: I've refactored `scroll_fn` to scroll on the window as opposed to the buffer, which has _another_ funny effect. This is relatively easy to make consistently.

Reply:
`nvim_win_call` works for me